web-marketplace/src/pages/Login/Login.tsx

```diff
@@ -0,0 +1,37 @@
+import { useEffect } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+import { Flex } from 'web-components/lib/components/box';
+import ConnectWallet from 'web-components/lib/components/organisms/ConnectWallet';
+
+import { useAuth } from 'lib/auth/auth';
+
+import { LoginButton } from 'components/organisms/LoginButton/LoginButton';
+
+import { LOGIN_PAGE_TITLE } from './Login.constants';
+import { useWallet } from 'lib/wallet/wallet';
+
+export const Login = () => {
+  const { loaded: walletLoaded, isConnected } = useWallet();
+  const { loading: authLoading, activeAccountId } = useAuth();
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+  const strict = searchParams.get('strict');
+
+  useEffect(() => {
+    if (!strict) {
+      if (!authLoading && activeAccountId || walletLoaded && isConnected) 
+        navigate(-1);
+    } else if (!authLoading && activeAccountId) navigate(-1);
+  }, [navigate, authLoading, activeAccountId]);
```

Comment:
```suggestion
    // needs Web2 or Web3 auth
    if (!strict) {
      if (!authLoading && activeAccountId || walletLoaded && isConnected) 
        navigate(-1);
    // needs necessarily Web2
    } else if (!authLoading && activeAccountId) navigate(-1);
  }, [navigate, authLoading, activeAccountId]);
```

Reply:
The `else` statement  is not about needing necessarily Web2 but rather a logged in user (whether it's done via web2 or web3), which is not the case for WC users.
I'll add some comments to clarify that